freestyle/shared/src/main/scala/freestyle/module.scala

```diff
@@ -171,8 +97,24 @@ final class moduleImpl(val c: Context) {
       Template(parentTrees, self, body0)).validNel
   }
 
+  private[this] def makeCoproductTree(effectTuples: List[(Name, Tree)]): Tree = {
```

Comment:
* ¿Can we do the `map (_._2)` in the call to `makeCoproductTree`? 
* ¿Can we narrow down the type of the `Tree` to what `.tpt` is?